moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -91,156 +129,161 @@ void JogCalcs::startMainLoop(JogArmShared& shared_variables)
-
-    // If paused or while waiting for initial jog commands, just keep the low-pass filters up to date with current
-    // joints so a jump doesn't occur when restarting
-    if (wait_for_jog_commands || shared_variables.paused)
-    {
-      for (std::size_t i = 0; i < num_joints_; ++i)
-        position_filters_[i].reset(original_joint_state_.position[i]);
+    if (latest_twist_stamped_)
+      twist_stamped_ = *latest_twist_stamped_;
+    if (latest_joint_jog_)
+      joint_jog_ = *latest_joint_jog_;
 
-      shared_variables.lock();
-      // Check if there are any new commands with valid timestamp
-      wait_for_jog_commands = shared_variables.command_deltas.header.stamp == ros::Time(0.) &&
-                              shared_variables.joint_command_deltas.header.stamp == ros::Time(0.);
-      shared_variables.unlock();
-    }
-    // If not waiting for initial command, and not paused.
-    // Do jogging calculations only if the robot should move, for efficiency
-    else
-    {
-      // Halt if the command is stale or inputs are all zero, or commands were zero
-      shared_variables.lock();
-      bool have_nonzero_cartesian_cmd = shared_variables.have_nonzero_cartesian_cmd;
-      bool have_nonzero_joint_cmd = shared_variables.have_nonzero_joint_cmd;
-      bool stale_command = shared_variables.command_is_stale;
-      shared_variables.unlock();
-
-      bool valid_nonzero_command = false;
-      if (!stale_command)
-      {
-        // Prioritize cartesian jogging above joint jogging
-        if (have_nonzero_cartesian_cmd)
-        {
-          shared_variables.lock();
-          cartesian_deltas = shared_variables.command_deltas;
-          shared_variables.unlock();
+    // Check for stale cmds
+    shared_variables_.command_is_stale =
+        ((ros::Time::now() - latest_command_stamp_) >= ros::Duration(parameters_.incoming_command_timeout));
+  }
 
-          if (!cartesianJogCalcs(cartesian_deltas, shared_variables))
-            continue;
-        }
-        else if (have_nonzero_joint_cmd)
-        {
-          shared_variables.lock();
-          joint_deltas = shared_variables.joint_command_deltas;
-          shared_variables.unlock();
+  // Get the transform from MoveIt planning frame to jogging command frame
+  // We solve (planning_frame -> base -> robot_link_command_frame)
+  // by computing (base->planning_frame)^-1 * (base->robot_link_command_frame)
+  tf_moveit_to_cmd_frame_ = kinematic_state_->getGlobalLinkTransform(parameters_.planning_frame).inverse() *
+                            kinematic_state_->getGlobalLinkTransform(parameters_.robot_link_command_frame);
 
-          if (!jointJogCalcs(joint_deltas, shared_variables))
-            continue;
-        }
+  // Input frame determined by YAML file if not passed with message
+  if (twist_stamped_.header.frame_id.empty())
+  {
+    twist_stamped_.header.frame_id = parameters_.robot_link_command_frame;
+  }
+  if (twist_stamped_.header.frame_id.empty())
```

Comment:
This if statement is the same as the one above it. I believe it was intended to be `if (joint_jog_.header.frame_id.empty())`, in which case I am not sure it is needed as jogging in the joint space does not have a frame or require a transformation

Reply:
Good catch, I'll remove this one.